### ISSUE TITLE
not matched result for InterMask

### ISSUE THREAD

**Issue by gengzichen** [gohar-malik/InterMask#3]
Dear Authors,

First thank you so much for your great work. I am now trying to follow your work but finding some unmatched results. 

========== MM Distance Summary ==========
---> [ground truth] Mean: 3.7853 CInterval: 0.0006
---> [best_fid.tar] Mean: 3.7913 CInterval: 0.0011
========== R_precision Summary ==========
---> [ground truth](top 1) Mean: 0.4165 CInt: 0.0050;(top 2) Mean: 0.6029 CInt: 0.0039;(top 3) Mean: 0.7034 CInt: 0.0050;
---> [best_fid.tar](top 1) Mean: 0.4586 CInt: 0.0059;(top 2) Mean: 0.6143 CInt: 0.0062;(top 3) Mean: 0.6956 CInt: 0.0066;
========== FID Summary ==========
---> [ground truth] Mean: 0.2979 CInterval: 0.0077
---> [best_fid.tar] Mean: 5.8878 CInterval: 0.1107
========== Diversity Summary ==========
---> [ground truth] Mean: 7.7664 CInterval: 0.0262
---> [best_fid.tar] Mean: 8.0339 CInterval: 0.0270
========== MultiModality Summary ==========
---> [best_fid.tar] Mean: 2.1719 CInterval: 0.0805

All the resulted **R-precisions are higher** than you claimed in the paper but the **FID is higher** than 5.154 in your claim, and **Multimodality** is higher than what you claimed. I strictly follow your work and there is no modification at all. Could you please give any suggestions on this?

Best Regards,
Zen

**Comment by jutanke-sony** [comment]
Jumping in on this thread - I can also only reproduce the results as reported by @gengzichen 

```
========== MM Distance Summary ==========
---> [ground truth] Mean: 3.7862 CInterval: 0.0008
---> [best_fid.tar] Mean: 3.7921 CInterval: 0.0013
========== R_precision Summary ==========
---> [ground truth](top 1) Mean: 0.4214 CInt: 0.0055;(top 2) Mean: 0.6044 CInt: 0.0052;(top 3) Mean: 0.7035 CInt: 0.0045;
---> [best_fid.tar](top 1) Mean: 0.4566 CInt: 0.0049;(top 2) Mean: 0.6146 CInt: 0.0044;(top 3) Mean: 0.6929 CInt: 0.0043;
========== FID Summary ==========
---> [ground truth] Mean: 0.2904 CInterval: 0.0062
---> [best_fid.tar] Mean: 5.8563 CInterval: 0.0663
========== Diversity Summary ==========
---> [ground truth] Mean: 7.7613 CInterval: 0.0285
---> [best_fid.tar] Mean: 8.0417 CInterval: 0.0291
========== MultiModality Summary ==========
---> [best_fid.tar] Mean: 2.2071 CInterval: 0.0869
```

**Comment by gengzichen** [comment]
A very interesting thing is that in the InterMask, the InterHuman Dataloader swap the motion 1 and motion 2 twice (in in2IN they only do this once), but this will somehow increase the FID from 0.269 to 0.285 or so. I think this is a very curious thing because probabilistically speaking this won't change the probability of the order of motion 1 and motion 2 (both 50 and 50). But this happens, anyone could help me with this lol.

**Comment by jutanke-sony** [comment]
> A very interesting thing is that in the InterMask, the InterHuman Dataloader swaps motion 1 and motion 2 twice (in in2IN they only do this once), but this will somehow increase the FID from 0.269 to 0.285 or so. This is a very curious thing because probabilistically speaking this won't change the probability of the order of motion 1 and motion 2 (both 50 and 50). But this happens, anyone could help me with this lol.

I would assume that this is due to stochasticity, as our reproduced numbers are close but there's still a gap. 


@gengzichen did you try reproducing the numbers from the original [InterGen](https://github.com/tr3e/InterGen), using their pre-trained model weights?

The results I can reproduce (not training InterGen, just using their pre-trained model weights) are also inconsistent with what has been reported there:
```
========== MM Distance Summary ==========
---> [ground truth] Mean: 3.7859 CInterval: 0.0006
---> [InterGen] Mean: 3.7998 CInterval: 0.0009
========== R_precision Summary ==========
---> [ground truth](top 1) Mean: 0.4226 CInt: 0.0051;(top 2) Mean: 0.6063 CInt: 0.0054;(top 3) Mean: 0.7048 CInt: 0.0057;
---> [InterGen](top 1) Mean: 0.4328 CInt: 0.0050;(top 2) Mean: 0.5886 CInt: 0.0041;(top 3) Mean: 0.6708 CInt: 0.0048;
========== FID Summary ==========
---> [ground truth] Mean: 0.2676 CInterval: 0.0065
---> [InterGen] Mean: 6.6189 CInterval: 0.0692
========== Diversity Summary ==========
---> [ground truth] Mean: 7.7506 CInterval: 0.0287
---> [InterGen] Mean: 7.8794 CInterval: 0.0250
========== MultiModality Summary ==========
---> [InterGen] Mean: 1.2286 CInterval: 0.0276
```
I wonder if there is some version shenanigans going on.. @gohar-malik have you experienced any of those issues or tried to reproduce InterGen results?

**Comment by gengzichen** [comment]
Thanks, but I guess this is probably not stochasticity. Besides, I haven't done experiments on InterGen, the only thing I can assume is that they have different versions of torch, which causes calculation different between tensors. Besides, there might be some really small bugs like MDM's R-precisions (they didn't add shuffle=True when constructing the test dataset)

**Comment by gohar-malik** [comment]
Hi @gengzichen, @jutanke-sony,

First off, thank you both for your interest in our work. Apologies for the delayed response; I’ve been away and just got back to catching up.

You’re absolutely right to point out the discrepancies. After checking, I realized that the original code and checkpoints were mistakenly uploaded from an earlier experiment that didn't reflect the final results reported in the paper.Apologies for the confusion this caused.

I’ve now updated both the code and the checkpoint link with the correct version from our final experiment. These should allow you to reproduce the results as reported.

I’ll go ahead and close this issue for now, but feel free to reopen it or tag me if you still run into any inconsistencies. Happy to help.

**Comment by gohar-malik** [comment]
Just to add a quick note, one small change in this release is the use of a few Graph Convolutional layers in the VQ-VAE instead of standard Conv2D layers. This was done to better capture the underlying pose structure, especially when working with global representations in the InterHuman dataset.

Also, while the updated code and checkpoint should give results very close to those reported in the paper, they might not match exactly (to tenth/hundredth decimal point). This is primarily because latent distribution metrics like FID can be quite sensitive to environment-level or even hardware-level variations (we have noticed different values when using different GPUs). To enable fair comparison, we’ve tried our best to match the evaluation setup and configurations used in [InterGen](https://github.com/tr3e/InterGen/tree/master). 

@jutanke-sony, I also faced the same issue when trying to reproduce InterGen's results. The metric values I obtained were similar to yours as opposed to what is reported in their paper. I am not quite sure what the issue is there. Hope this helps clarify things!

**Comment by catherineytw** [comment]
Thank you for the awesome work and codes. I ran the evaluation code using the latest checkpoints and finally got the following results. The R precisions are almost the same as you claimed in the paper, but the FIDs are higher than 5.15 in your paper, and multimodalities are higher than 1.737 in your paper. Are these results correct? I didn't modify any of the codes. 


========== MM Distance Summary ==========
---> [ground truth] Mean: 3.7845 CInterval: 0.0007
---> [best_fid.tar] Mean: 3.7910 CInterval: 0.0013
========== R_precision Summary ==========
---> [ground truth](top 1) Mean: 0.4235 CInt: 0.0050;(top 2) Mean: 0.6023 CInt: 0.0050;(top 3) Mean: 0.7028 CInt: 0.0059;
---> [best_fid.tar](top 1) Mean: 0.4469 CInt: 0.0052;(top 2) Mean: 0.5989 CInt: 0.0043;(top 3) Mean: 0.6822 CInt: 0.0046;
========== FID Summary ==========
---> [ground truth] Mean: 0.2915 CInterval: 0.0054
---> [best_fid.tar] Mean: 5.2182 CInterval: 0.0657
========== Diversity Summary ==========
---> [ground truth] Mean: 7.7896 CInterval: 0.0227
---> [best_fid.tar] Mean: 7.9990 CInterval: 0.0328
========== MultiModality Summary ==========
---> [best_fid.tar] Mean: 2.3386 CInterval: 0.0673

Best Regards,
Wang
